### PR TITLE
Feat/ping 1480 auto welcome msg

### PR DESCRIPTION
**Pull request by brm-stnd**
Saat ini untuk pengiriman pesan dan delay sudah selesai tinggal exclude use yg pernah menerima pesan yang blm

### CODE REVIEW

File: src/processes/auto-welcome-msg-process.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      Bastian`,
+        mentioned_users: [resultPrepopulated.ownUser.user_id]
+      };
+
+      await chat.sendMessage({
+        ...toBeSent,
+        user_id: resultPrepopulated.targetUser.user_id
+      });
+    }
+
```
Comment (fajarism) [inline src/processes/auto-welcome-msg-process.js:42]: Please add chat.stopWatching() after sending message so stream connection can be closed.
https://getstream.io/chat/docs/react/watch_channel/#stop-watching-a-channel

File: src/processes/auto-welcome-msg-process.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+const automateWelcomeMsgProcess = async (job, done) => {
+  const serverClient = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+  try {
+    let data = job.data;
+
+    const resultPrepopulated = data.resultPrepopulated[0];
+    const channelId = resultPrepopulated.generatedChannelId;
+
+    console.log('channelId');
+    console.log(channelId);
```
Comment (fajarism) [inline src/processes/auto-welcome-msg-process.js:12]: Remove unnecessary console.log

File: src/processes/helper/preAutoWelcomeMsg.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    if (parseInt(getCurrentTimeHour) > randomTime) {
+      additionalDays = 1;
+    }
+    let requiredTime = momentTz()
+      .tz('America/Los_Angeles')
+      .set({hour: randomTime})
+      .add(additionalDays, 'days');
+
+    console.log('currentTime', currentTime);
+    console.log('requiredTime', requiredTime);
```
Comment (fajarism) [inline src/processes/helper/preAutoWelcomeMsg.js:22]: Remove unnecessary console log

File: src/services/chat/prepopulated.js [hunk 4]
```diff
@@ -1,84 +1,88 @@
...
-             * boleh tampil kecuali untuk user usup
-             */
+    let res = await Promise.all(
+      users.map(async (user) => {
+        const channelName = [];
+        channelName.push(user.username);
+        channelName.push(ownUser.username);
+        let generatedChannelId = generateRandomId();
+        console.log('generatedChannelId');
+        console.log(generatedChannelId);
```
Comment (fajarism) [inline src/services/chat/prepopulated.js:21]: Remove unnecessary console log

**Review by fajarism — CHANGES_REQUESTED** [review]
Summary:
1. Please add chat.stopWatching()
2. Remove unnecessary console log

File: src/processes/auto-welcome-msg-process.js (hunk 1 above)
Comment (brm-stnd) [reply src/processes/auto-welcome-msg-process.js]: fixed

File: src/processes/auto-welcome-msg-process.js (hunk 2 above)
Comment (brm-stnd) [reply src/processes/auto-welcome-msg-process.js:12]: i'll remove later after test result was correct

File: src/processes/helper/preAutoWelcomeMsg.js (hunk 3 above)
Comment (brm-stnd) [reply src/processes/helper/preAutoWelcomeMsg.js:22]: i'll remove later after test result was correct

File: src/services/chat/prepopulated.js (hunk 4 above)
Comment (brm-stnd) [reply src/services/chat/prepopulated.js:21]: i'll remove later after test result was correct

**Review by fajarism — APPROVED** [review]

**Pull request merged**